### PR TITLE
New geometry class

### PR DESCRIPTION
**Pull request by code-explorer**
### :pencil: Description

**Type:** ::rocket: `restructure` 

New geometry class for model state restructure.

### :pushpin: Resources

Model State Plan

### CODE REVIEW

File: tardis/model/base.py [hunk 1]
```diff
@@ -84,6 +84,26 @@ def r_outer(self):
         return self.geometry.r_outer.values * self.geometry_units["r_outer"]
 
 
+class Geometry:
+    """
+    Holds information about model geometry
+
+    Parameters
+    ----------
+    radius : astropy.units.quantity.Quantity
```
Comment (Rodot-) [inline tardis/model/base.py:93]: Give this the properties `r_inner`, `r_outer`, `v_inner`, and `v_outer` as numpy arrays containing units

File: tardis/model/base.py [hunk 2]
```diff
@@ -84,6 +84,26 @@ def r_outer(self):
         return self.geometry.r_outer.values * self.geometry_units["r_outer"]
 
 
+class Geometry:
```
Comment (Rodot-) [inline tardis/model/base.py:87]: For now, just make a class called `Radial1DGeometry` which will eventually inherit from a Geometry parent class.  The `Radial1DGeometry` should contain the attributes needed to run tardis with a radial 1D geometry

File: tardis/model/base.py (hunk 1 above)
Comment (Rodot-) [reply tardis/model/base.py:93]: Make sure it has methods to compute volumes in each shell

**Review by Rodot- — CHANGES_REQUESTED** [review]

File: tardis/model/base.py [hunk 3]
```diff
@@ -84,6 +84,47 @@ def r_outer(self):
         return self.geometry.r_outer.values * self.geometry_units["r_outer"]
 
 
+class Radial1DGeometry:
+    """
+    Holds information about model geometry
```
Comment (andrewfullard) [inline tardis/model/base.py:89]: Be specific about the geometry type

File: tardis/model/base.py [hunk 4]
```diff
@@ -84,6 +84,47 @@ def r_outer(self):
...
+
+    Parameters
+    ----------
+    radius : astropy.units.quantity.Quantity
+    velocity : astropy.units.quantity.Quantity
+    """
+
+    def __init__(self, radius, velocity):
+        self.geometry_df = pd.DataFrame(
+            {"radius": radius, "velocity": velocity}
```
Comment (andrewfullard) [inline tardis/model/base.py:99]: Dataframes can't read astropy quantities, so they need to be explicitly set to CGS units with the `.to_cgs` method and then converted to a pure numpy array with the `.value` method.

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: tardis/model/base.py [hunk 5]
```diff
@@ -84,6 +84,47 @@ def r_outer(self):
...
+    Holds information about model geometry
+
+    Parameters
+    ----------
+    radius : astropy.units.quantity.Quantity
+    velocity : astropy.units.quantity.Quantity
+    """
+
+    def __init__(self, radius, velocity):
+        self.geometry_df = pd.DataFrame(
```
[the remark this answers is not in the document]
Comment (andrewfullard) [reply tardis/model/base.py:98]: To be clear, we have decided to cut the dataframe step. Then the class can just have `self.radius` and `self.velocity`. The properties can remain, they are just created without the dataframe. `self.radius` and `self.velocity` can keep their astropy quantities as a result, you can just use `to(u.cm)` for example.

File: tardis/model/base.py [hunk 6]
```diff
@@ -84,6 +84,41 @@ def r_outer(self):
...
+    """
+    Holds information about model geometry for radial 1D models.
+
+    Parameters
+    ----------
+    radius : astropy.units.quantity.Quantity
+    velocity : astropy.units.quantity.Quantity
+    """
+
+    def __init__(self, radius, velocity):
```
Comment (wkerzendorf) [inline tardis/model/base.py:97]: the call signature should be `def __init__(self, r_inner, r_outer, v_inner, v_outer):`

**Review by andrewfullard — APPROVED** [review]

File: tardis/model/base.py [hunk 7]
```diff
@@ -84,6 +84,27 @@ def r_outer(self):
         return self.geometry.r_outer.values * self.geometry_units["r_outer"]
 
 
+class Radial1DGeometry:
+    """
+    Holds information about model geometry for radial 1D models.
+
+    Parameters
+    ----------
+    radius : astropy.units.quantity.Quantity
```
Comment (Rodot-) [inline tardis/model/base.py:93]: Make sure the description of the doc-string matches the parameters of the __init__

**Review by AlexHls — DISMISSED** [review]

**Review by andrewfullard — APPROVED** [review]

**Review by Rodot- — DISMISSED** [review]

**Review by wkerzendorf — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Review by AlexHls — APPROVED** [review]

**Review by Rodot- — APPROVED** [review]

**Pull request merged**